TSS.NET/TSS.Net.UWP/CryptoAsym.cs

```diff
@@ -0,0 +1,1010 @@
+                    Globs.Throw<ArgumentException>("Key scheme and signature scheme do not match");
+                    return false;
+                }
+
+                byte[] digest = dataIsDigest ? data : CryptoLib.HashData(sigHash, data);
+
+                if (sigScheme == TpmAlgId.Rsassa)
+                {
+                    return Key.VerifySignature(digest, s.sig, sigHash, true);
+                }
+                if (sigScheme == TpmAlgId.Rsapss)
+                {
+                    Globs.Throw<ArgumentException>("VerifySignature(): PSS scheme is not supported");
+                    return false;
+                }
+                Globs.Throw<ArgumentException>("VerifySignature(): Unrecognized scheme");
+                return false;
+            }
+
+            var eccParams = PublicParms.parameters as EccParms;
+            if (eccParams != null)
+            {
+                if (eccParams.scheme.GetUnionSelector() != TpmAlgId.Ecdsa)
+                {
+                    Globs.Throw<ArgumentException>("Unsupported ECC sig scheme");
+                    return false;
+                }
+                TpmAlgId keyScheme = eccParams.scheme.GetUnionSelector();
+
+                if (keyScheme != TpmAlgId.Null && keyScheme != sigScheme)
+                {
+                    Globs.Throw<ArgumentException>("Key scheme and signature scheme do not match");
+                    return false;
+                }
+
+                var s = sig as SignatureEcdsa;
+                byte[] digest = dataIsDigest ? data : CryptoLib.HashData(sigHash, data);
+                byte[] sigBlob = Globs.Concatenate(s.signatureR, s.signatureS);
+                return Key.VerifySignature(digest, sigBlob);
+            }
+
+            // Should never be here
+            Globs.Throw("VerifySignature: Unrecognized asymmetric algorithm");
+            return false;
+        } // VerifySignature()
+
+        /// <summary>
+        /// Generates the key exchange key and the public part of the ephemeral key
+        /// using specified encoding parameters in the KDF (ECC only).
+        /// </summary>
+        /// <param name="encodingParms"></param>
+        /// <param name="decryptKeyNameAlg"></param>
+        /// <param name="ephemPub"></param>
+        /// <returns>key exchange key blob</returns>
+        public byte[] EcdhGetKeyExchangeKey(byte[] encodingParms, TpmAlgId decryptKeyNameAlg, out EccPoint ephemPub)
+        {
+            byte[] keyExchangeKey = null;
+            ephemPub = null;
+
+            var eccParms = (EccParms)PublicParms.parameters;
+            int keyBits = RawEccKey.GetKeyLength(eccParms.curveID);
+
+            // Make a new ephemeral key
+            var ephKey = Generate(RawEccKey.GetEccAlg(PublicParms), (uint)keyBits);
+            byte[] ephPub = ephKey.Export(Native.BCRYPT_ECCPUBLIC_BLOB);
+            byte[] otherPub = Key.Export(Native.BCRYPT_ECCPUBLIC_BLOB);
+
+            byte[] herPubX, herPubY;

```

Comment:
Thanks for fixing the formatting here.

Reply:
You bet! Turns out formatting is a lot easier without a lot of inline `#ifdefs` that introduce scopes...